apis/exoscale/v1/dbaas_types.go

```diff
@@ -0,0 +1,138 @@
+package v1
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	exoscaleoapi "github.com/exoscale/egoscale/v2/oapi"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// NodeState describes the state of a service node.
+type NodeState struct {
+	// Name of the service node
+	Name string `json:"name,omitempty"`
+	// Role of this node.
+	Role exoscaleoapi.DbaasNodeStateRole `json:"role,omitempty"`
+	// State of the service node.
+	State exoscaleoapi.DbaasNodeStateState `json:"state,omitempty"`
+}
+
+// Notification contains a service message.
+type Notification struct {
+	// Level of the notification.
+	Level exoscaleoapi.DbaasServiceNotificationLevel `json:"level,omitempty"`
+	// Message contains the notification.
+	Message string `json:"message,omitempty"`
+	// Type of the notification.
+	Type exoscaleoapi.DbaasServiceNotificationType `json:"type,omitempty"`
+	// Metadata contains additional data.
+	Metadata runtime.RawExtension `json:"metadata,omitempty"`
+}
+
+// MaintenanceSpec contains settings to control the maintenance of an instance.
+type MaintenanceSpec struct {
+	// DayOfWeek specifies at which weekday the maintenance is held place.
```

Comment:
Maybe add the comment for kubebuilder to handle this as an enum?

Reply:
At first I wanted to avoid having a mix of kubebuilder validation and webhooks. I'd rather have only 1 place for validation. But for enums the validation code is actually a pain to have, e.g. using switch cases. So yeah, I can add that.

Reply:
TBH I'd go the path of least resistance. If there's a very complicated verification -> webhook. If it can be handled built-in provided by kubebuilder -> kubebuilder. No need to re-invent the wheel...